text/metrics/0001-naming-conventions.md

```diff
@@ -0,0 +1,147 @@
+
+`system.cpu` may include labels for per-core measures:
+* `cpu.core.[0-n]`, eg, `cpu.core.3`
+
+Cores should be reported ordinally as ordered by the operating system. It is recommended that values begin at 0.
+
+It is recommended that per-core labels should not be reported by default to reduce cardinality; a user should opt-in with via configuration.
+
+#### Network
+
+`system.network` is the core metric name.
+
+Standard labels include, non-exhaustively,
+* `network.sent`
+* `network.received`
+
+`system.network` may include labels for per-NIC measures:
+* `network.nic.[0-n]`, eg, `network.nic.3`
+
+NICs should be reported ordinally as ordered by the operating system. It is recommended that values begin at 0.
+
+Interfaces may also be reported by OS name, eg, `en3` in the label `network.nic.en3`.
+
+It is recommended that per-NIC labels should not be reported by default to reduce cardinality; a user should opt-in with via configuration.
+
+TODO: what other network labels? dropped? how about low-level things like window size?
+
+#### Memory
+
+`system.memory` is the core metric name.
+
+Standard labels include, non-exhaustively,
+* `memory.free`
+* `memory.resident`
+* `memory.shared`
+* `memory.private`
+
+TODO: how can a user derive total memory? how about memory utilization? memory allocations that are reported in more than one label may make this difficult.
+
+#### More system metrics
+
+Possibilities:
+* Disk
+* Filesystems
+* Load
+* Per-process
+* Processes and threads?
+
+Note for discussion: see this [excellent reference guide](https://docs.google.com/spreadsheets/d/11qSmzD9e7PnzaJPYRFdkkKbjTLrAKmvyQpjBjpJsR2s/edit#gid=0)
+
+### Caution
+
+Operating systems will report different labels for common metrics based on their architecture. Queries should be scoped to a host or cluster running the same operating system to avoid aggregating dissimilar measures.
+
+
+#### TODO lower-level system metrics to consider
+* CPU interrupts
+* System calls
+* Swap and paging
+
+### Application Runtime
+
+All runtime metric names should be reported with a namespace that includes the name of the runtime, eg, `runtime.go.*`.
+
+#### Go
+
+All Go runtime metrics should be within the `runtime.go.*` namespace.
+
+Common metrics include:
+* `runtime.go.goroutines`
+* `runtime.go.heap_alloc`
+* `runtime.go.gc` with labels `gc.count`
+
+#### Java
+
+All Java runtime metrics should be within the `runtime.java.*` namespace.
+
+* `runtime.java.threads` with optional labels for thread pools, eg, `runtime.java.thread_pool.[name]`
+* `runtime.java.heap_alloc`
+* `runtime.java.gc` with labels `gc.count` and `gc.time`
+
+#### Node
+
+All Node runtime metrics should be within the `runtime.node.*` namespace.
+
+* `runtime.node.gc` with labels `pause.time` and `pause.count`
+* `runtime.node.heap_alloc` with labels `heap.total_size`, `heap.available_size`, `heap.used_heap_size` **TODO** confirm
+* `runtime.node.event_loop` with TBD labels **TODO**
```

Comment:
Please use `nodejs` :)) `Node` is too generic and may confuse with kubernets.node :))

Reply:
63443ef19282b4764b130eece22fd2082f635780